ios/Classes/TiNfcVTagTechnology.m

```diff
@@ -9,4 +9,701 @@
+                                                keyIdentifier:[keyIdentifier integerValue]
+                                                      message:data
+                                            completionHandler:^(NFCISO15693ResponseFlag responseFlag, NSData *_Nonnull response, NSError *_Nullable error) {
+                                              if (![self _hasListeners:@"didKeyUpdateWithRequestFlags"]) {
+                                                return;
+                                              }
+                                              [self fireEvent:@"didKeyUpdateWithRequestFlags"
+                                                   withObject:@{
+                                                     @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                     @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                     @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                   }];
+                                            }];
+}
+
+- (void)readBufferWithRequestFlags:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  [[self.tagProxy asNFCISO15693Tag] readBufferWithRequestFlags:requestFlags
+                                             completionHandler:^(NFCISO15693ResponseFlag responseFlag, NSData *_Nonnull data, NSError *_Nullable error) {
+                                               if (![self _hasListeners:@"didReadBufferWithRequestFlags"]) {
+                                                 return;
+                                               }
+                                               [self fireEvent:@"didReadBufferWithRequestFlags"
+                                                    withObject:@{
+                                                      @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                      @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                      @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                    }];
+                                             }];
+}
+
+- (void)sendRequestWithFlag:(id)args
+{
+  NSNumber *flag = [[args firstObject] valueForKey:@"flag"];
+  NSNumber *commandCode = [[args firstObject] valueForKey:@"commandCode"];
+
+  NSArray *commandData = [[args firstObject] valueForKey:@"data"];
+  unsigned int dataValue[commandData.count];
+  for (int i = 0; i < commandData.count; i++) {
+    dataValue[i] = [commandData[i] unsignedIntValue];
+  }
+  NSData *data = [[NSData alloc] initWithBytes:dataValue length:commandData.count];
```

Comment:
Same here, since it is repetitive code block, better to create a separate methods with name like **arrayToData**.

Reply:
Method Added